### ISSUE TITLE
Smeltery requests a ton of fuel

### ISSUE THREAD

**Issue by Paril** [ldtteam/minecolonies#3776]
https://i.imgur.com/jxM9hZ6.png

He will continuously fill up his request queue for Coal and Charcoal, both of which I specified him to use.

Version is minecolonies-1.12.2-0.10.337-ALPHA-universal.jar

**Comment by WenXin20** [comment]
I've noticed that the same thing also happens with the cook.

**Comment by Paril** [comment]
I just had an "aha" thought: is it possible the Smelter/Cook is requesting the max amount of stacks for fuel for the amount of furnaces the Level 5 building has?

**Comment by WenXin20** [comment]
Probably not, my Cook is only a level 2, and the Smelter a level 1, but I still see a lot of those requests.

**Comment by marchermans** [comment]
They can only request 1 stack of fuel at a time.
But they will keep on requesting if they think there is more space, which there usually is.

**Comment by Paril** [comment]
Yeah I was mistaken in that it's Fuel he was requesting, not individual stacks of both, but like today he had 40+ requests of it. He should only request one at a time ever, because it's just gonna keep growing.

I've also noticed his Smeltable Ores request get stuck a lot. There's ore in the warehouse, but it might have been put in after or something because it was never fulfilled. I had to cancel the request to get him to refresh it. I thought maybe it was because he had fulfilled his daily quota, but many days later he was still requesting it. Immediately after cancelling, he put his request in again and it was fulfilled with 4 responses by the warehouse. Maybe when a worker wakes up and reaches his workplace, if he has unfulfilled requests they should be soft refreshed to see if they can be resolved again. 

Oddly, my cook doesn't request fuel. He has a stack of coal and he's content with that. Meanwhile the smelter thinks he needs several boatloads of fuel for no reason.

I've also had some odd occurrences with the Miner silently requesting cobblestone slabs. There was no request, but once I gave them to her she went back to work immediately.

**Comment by Raycoms** [comment]
I'll check it out, that should've been fixed in the last version.

Am Mo., 5. Aug. 2019 um 17:20 Uhr schrieb Jonathan <notifications@github.com

**Comment by marchermans** [comment]
Hmm that looks like detection is broken, but only on the smelter, that would be interesting since they run through the same codepath, for requesting items, @Raycoms did you change that?

**Comment by trials117** [comment]
Just wanted to say I am having the exact same problem with my cook (Haven't played Mine colonies in a year and not built a smelter yet) 

Will continuously request for stacks of fuel (Spruce planks) from the player and not the store. It sends so many requests the game gets lagged and opening the colony clipboard or selecting the cook's requests would fairly consistently cause a crash. Turning off the spruce planks as fuel did not stop the requests and they had to be manually cancelled. 

Something I found interesting is that the cook did not start cooking until I had placed fuel in the furnace, despite cook having fuel in inventory and in the restaurant's inventory
